test/test_loss.py

```diff
@@ -32,12 +32,11 @@ def test_ctc_loss():
     ch_loss = F.connectionist_temporal_classification(
         ch_pred, ch_target, 0, input_length, label_length).data
 
-    th_pred = pad_list([torch.autograd.Variable(torch.from_numpy(x))
+    th_pred = pad_list([torch.from_numpy(x)
                         for x in np_pred], 0.0).transpose(0, 1)
-    th_target = torch.autograd.Variable(
-        torch.from_numpy(numpy.concatenate(np_target)))
-    th_ilen = torch.autograd.Variable(torch.from_numpy(input_length))
-    th_olen = torch.autograd.Variable(torch.from_numpy(label_length))
+    th_target = torch.from_numpy(numpy.concatenate(np_target))
+    th_ilen = torch.from_numpy(input_length)
+    th_olen = torch.from_numpy(label_length)
     # NOTE: warpctc_pytorch.CTCLoss does not normalize itself by batch-size while chainer's default setting does
     th_loss = (CTCLoss()(th_pred, th_target, th_ilen,
```

Comment:
use CTCLoss(size_averaged=True) instead

Reply:
maybe adding gradient equality check is nice. the different gradient was the reason why we used custom function? https://github.com/jnishi/warp-ctc/commit/3fd4783497bc1791034679ba70d1c74efc4a4468

Reply:
also check this https://github.com/espnet/espnet/pull/307

Reply:
For example wrapping with 
``` python
class SquareGrad(Function):
    @staticmethod
    def forward(ctx, input):
        return x
    @staticmethod
    def backward(ctx, grad_output):
        return grad_output * grad_output

def chainer_like_ctc(y_hat, y_true, ilens, olens):
    return warp_ctc.CTCLoss(size_average=True)(SquareGrad.apply(y_hat), y_true, ilens, olens)
```

Reply:
Thanks @ShigekiKarita.
Here the return value `x` in forward is `input`, right?